src/assets/wise5/components/multipleChoice/multiple-choice-student/multiple-choice-student.component.ts

```diff
@@ -0,0 +1,493 @@
+      }
+
+      const submitCounter = studentData.submitCounter;
+      if (submitCounter != null) {
+        this.submitCounter = submitCounter;
+      }
+
+      this.processLatestStudentWork();
+    }
+  }
+
+  setIsCheckedOnStudentChoices(studentChoices: string[]): void {
+    for (const choice of this.choices) {
+      if (studentChoices.includes(choice.id)) {
+        choice.isChecked = true;
+      }
+    }
+  }
+
+  showFeedbackForChoiceIds(choiceIds: string[]): void {
+    for (const choiceId of choiceIds) {
+      const choiceObject = this.getChoiceById(choiceId);
+      if (choiceObject != null) {
+        choiceObject.showFeedback = true;
+        choiceObject.feedbackToShow = choiceObject.feedback;
+      }
+    }
+  }
+
+  isChecked(choiceId: string): boolean {
+    const studentChoices = this.studentChoices;
+    if (studentChoices != null) {
+      if (this.isRadio()) {
+        if (choiceId === studentChoices) {
+          return true;
+        }
+      } else if (this.isCheckbox()) {
+        if (studentChoices.indexOf(choiceId) != -1) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  getChoiceIdsFromStudentData(studentData: any): string[] {
+    const choiceIds = [];
+    if (studentData != null && studentData.studentChoices != null) {
+      const studentChoices = studentData.studentChoices;
+      for (const studentDataChoice of studentChoices) {
+        if (studentDataChoice != null) {
+          const studentDataChoiceId = studentDataChoice.id;
+          choiceIds.push(studentDataChoiceId);
+        }
+      }
+    }
+    return choiceIds;
+  }
+
+  radioChoiceSelected(choiceId: string): void {
+    if (this.isDisabled) {
+      return;
+    }
+    this.studentDataChanged();
+    if (this.mode === 'student') {
+      const category = 'StudentInteraction';
+      const event = 'choiceSelected';
+      const data: any = {
+        selectedChoiceId: choiceId
+      };
+      this.StudentDataService.saveComponentEvent(this, category, event, data);
+    }
+  }
+
+  checkboxClicked(choiceId: string): void {
+    if (this.isDisabled) {
+      return;
+    }
+    this.addOrRemoveFromStudentChoices(choiceId);
+    this.studentDataChanged();
+    if (this.mode === 'student') {
+      const category = 'StudentInteraction';
+      const event = 'choiceSelected';
+      const data: any = {
+        selectedChoiceId: choiceId,
+        choicesAfter: this.studentChoices
+      };
+      this.StudentDataService.saveComponentEvent(this, category, event, data);
+    }
+  }
+
+  addOrRemoveFromStudentChoices(choiceId: string): void {
+    const index = this.studentChoices.indexOf(choiceId);
+    if (index == -1) {
+      // the choice was not previously checked so we will add the choice id to the array
+      this.studentChoices.push(choiceId);
+    } else {
+      // the choice was previously checked so we will remove the choice id from the array
+      this.studentChoices.splice(index, 1);
+    }
```

Comment:
```suggestion
    if (index === -1) {
      this.studentChoices.push(choiceId);
    } else {
      this.studentChoices.splice(index, 1);
    }
```
This code makes sense without the comments.